### PR TITLE
[Core] Adding python wheels scripts

### PR DESCRIPTION
**Pull request by HubertBalcerzak**
Scripts for python wheel generation.

### CODE REVIEW

**Comment by loumalouomega** [comment]
Just to mention that staring today we are compatible with MinGW, which means we can do cross compiling for Windows from Linux

File: scripts/wheels/linux/configure.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+-DCONSTITUTIVE_MODELS_APPLICATION=OFF                                               \
+-DFLUID_DYNAMICS_APPLICATION=ON                                                     \
+-DMESH_MOVING_APPLICATION=OFF                                                       \
+-DFSI_APPLICATION=OFF                                                               \
+-DDEM_APPLICATION=ON                                                                \
+-DSWIMMING_DEM_APPLICATION=OFF                                                      \
+-DMIXED_ELEMENT_APPLICATION=OFF                                                     \
+-DSHAPE_OPTIMIZATION_APPLICATION=OFF                                                \
+-DTOPOLOGY_OPTIMIZATION_APPLICATION=OFF                                             \
+-DCONTACT_STRUCTURAL_MECHANICS_APPLICATION=ON                                       \
```
Comment (loumalouomega) [inline scripts/wheels/linux/configure.sh:25]: :)

File: scripts/wheels/linux/setup.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return True
+
+
+class EmptyListWithLength(list):
+    def __len__(self):
+        return 1
+
+setuptools.setup(
+    name="KratosMultiphysicsWheelTest",
+    version="0.1",
```
Comment (loumalouomega) [inline scripts/wheels/linux/setup.py]: I would use the official version (7.0+hash(last master commit))

File: scripts/wheels/windows/setup.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#     long_description = fh.read()
+
+class BinaryDistribution(setuptools.Distribution):
+    def has_ext_modules(foo):
+        return True
+
+
+setuptools.setup(
+    name="KratosMultiphysicsWheelTest",
+    version="0.1",
```
Comment (loumalouomega) [inline scripts/wheels/windows/setup.py]: Idem

**Review by loumalouomega — COMMENTED** [review]
@Hubertus248 send me a e-mail to vmataix@cimne.upc.edu, I would send you the login info for the "official" Kratos pip

**Comment by HubertBalcerzak** [comment]
> 
> 
> Just to mention that staring today we are compatible with MinGW, which means we can do cross compiling for Windows from Linux

We need windows anyway to generate wheel files

File: scripts/wheels/linux/setup.py (hunk 2 above)
Comment (HubertBalcerzak) [reply scripts/wheels/linux/setup.py]: done

File: scripts/wheels/windows/setup.py (hunk 3 above)
Comment (HubertBalcerzak) [reply scripts/wheels/windows/setup.py]: also done

File: scripts/wheels/linux/configure.sh (hunk 1 above)
Comment (HubertBalcerzak) [reply scripts/wheels/linux/configure.sh:25]: I'm not sure what's the suggestion here

Comment (loumalouomega) [reply scripts/wheels/linux/configure.sh:25]: I am happy, it is not a suggestion

**Comment by loumalouomega** [comment]
> > Just to mention that staring today we are compatible with MinGW, which means we can do cross compiling for Windows from Linux
> 
> We need windows anyway to generate wheel files

Not necessarily, you can install the MinGW support for gcc and compile it from linux: https://stackoverflow.com/questions/15986715/how-do-i-invoke-the-mingw-cross-compiler-on-linux

File: scripts/wheels/linux/configure.sh (hunk 1 above)
Comment (HubertBalcerzak) [reply scripts/wheels/linux/configure.sh:25]: :)

**Comment by HubertBalcerzak** [comment]
> 
> 
> > > Just to mention that staring today we are compatible with MinGW, which means we can do cross compiling for Windows from Linux
> > 
> > 
> > We need windows anyway to generate wheel files
> 
> Not necessarily, you can install the MinGW support for gcc and compile it from linux: https://stackoverflow.com/questions/15986715/how-do-i-invoke-the-mingw-cross-compiler-on-linux

Yes, but the wheel generation itself (`python setup.py bdist_wheel`) must be run on the target platform

**Comment by loumalouomega** [comment]
> > > > Just to mention that staring today we are compatible with MinGW, which means we can do cross compiling for Windows from Linux
> > > 
> > > 
> > > We need windows anyway to generate wheel files
> > 
> > 
> > Not necessarily, you can install the MinGW support for gcc and compile it from linux: https://stackoverflow.com/questions/15986715/how-do-i-invoke-the-mingw-cross-compiler-on-linux
> 
> Yes, but the wheel generation itself (`python setup.py bdist_wheel`) must be run on the target platform

Huuuuuuuuum...

File: scripts/wheels/linux/__init__.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
```
Comment (philbucher) [inline scripts/wheels/linux/__init__.py:1]: @Hubertus248 we did some important changes here in the last days, please use the latest __init__.py from the master

File: scripts/wheels/windows/__init__.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
```
Comment (philbucher) [inline scripts/wheels/windows/__init__.py:1]: Same

File: scripts/wheels/linux/__init__.py (hunk 4 above)
Comment (HubertBalcerzak) [reply scripts/wheels/linux/__init__.py:1]: done

File: scripts/wheels/windows/__init__.py (hunk 5 above)
Comment (HubertBalcerzak) [reply scripts/wheels/windows/__init__.py:1]: also done

File: scripts/wheels/linux/__init__.py (hunk 4 above)
Comment (philbucher) [reply scripts/wheels/linux/__init__.py:1]: I think sth went wrong, it is not the latest one:
https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_interface/__init__.py

Comment (HubertBalcerzak) [reply scripts/wheels/linux/__init__.py:1]: Yep, that was weird. Should be fine now

**Comment by HubertBalcerzak** [comment]
@loumalouomega @philbucher Is anything still blocking this?

**Review by loumalouomega — APPROVED** [review]
<3

imagen

**Comment by philbucher** [comment]
:+1:

**Pull request merged**